Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -838,6 +1104,295 @@ void MatchTOF::doMatching(int sec)
   return;
 }
 //______________________________________________
+void MatchTOF::doMatchingForTPC(int sec)
+{
+  printf("here, DoMatch\n");
+  auto& gasParam = o2::tpc::ParameterGas::Instance();
+  float vdrift = gasParam.DriftV;
+
+  ///< do the real matching per sector
+  mMatchedTracksPairs.clear(); // new sector
+
+  auto& cacheTOF = mTOFClusSectIndexCache[sec]; // array of cached TOF cluster indices for this sector; reminder: they are ordered in time!
+  auto& cacheTrk = mTracksSectIndexCache[sec];  // array of cached tracks indices for this sector; reminder: they are ordered in time!
+  int nTracks = cacheTrk.size(), nTOFCls = cacheTOF.size();
+  LOG(INFO) << "Matching sector " << sec << ": number of tracks: " << nTracks << ", number of TOF clusters: " << nTOFCls;
+  if (!nTracks || !nTOFCls) {
+    return;
+  }
+  int itof0 = 0; // starting index in TOF clusters for matching of the track
+  float deltaPosTemp[3];
+  std::array<float, 3> pos;
+  std::array<float, 3> posBeforeProp;
+  float posFloat[3];
+
+  // prematching for TPC only tracks (identify BC candidate to correct z for TPC track accordingly to v_drift)
+
+  std::vector<unsigned long> BCcand;
+
+  std::vector<int> nStripsCrossedInPropagation;
+  std::vector<std::array<std::array<int, 5>, 2>> detId;
+  std::vector<std::array<o2::track::TrackLTIntegral, 2>> trkLTInt;
+  std::vector<std::array<std::array<float, 3>, 2>> deltaPos;
+  std::vector<std::array<int, 2>> nStepsInsideSameStrip;
+
+  LOG(DEBUG) << "Trying to match %d tracks" << cacheTrk.size();
+  for (int itrk = 0; itrk < cacheTrk.size(); itrk++) {
+    auto& trackWork = mTracksWork[cacheTrk[itrk]];
+    auto& trefTrk = trackWork.first;
+    auto& intLT = mLTinfos[cacheTrk[itrk]];
+
+    int nBCcand = 1;
+    BCcand.clear();
+    nStripsCrossedInPropagation.clear();
+
+    int side = mSideTPC[cacheTrk[itrk]];
+
+    // look at BC candidates for the track
+    itof0 = 0;
+    double minTrkTime = (trackWork.second.getTimeStamp() - trackWork.second.getTimeStampError()) * 1.E6; // minimum time in ps
+    double maxTrkTime = (trackWork.second.getTimeStamp() + mExtraTPCFwdTime[cacheTrk[itrk]]) * 1.E6;     // maximum time in ps
```

Comment:
One should account for some margin due to the track time precision, I would add a fixed value of a ~5 TPC bins, i.e. 1ms.

Reply:
Done

Reply:
Margin add directly in prepareTPCTrack so that these lines are not changed.
See 
   timeInfo.setTimeStampError((trcOrig.getDeltaTBwd() + 5) * o2::tpc::ParameterElectronics::Instance().ZbinWidth);
   mExtraTPCFwdTime.push_back((trcOrig.getDeltaTFwd() + 5) * o2::tpc::ParameterElectronics::Instance().ZbinWidth);
at line 574

Reply:
@shahor02 just to be sure:


> One should account for some margin due to the track time precision, I would add a fixed value of a ~5 TPC bins, i.e. 1ms.

5 TPC time bins is about 1 microsecond, right? Since o2::tpc::ParameterElectronics::Instance().ZbinWidth is ~0.2 us.